numba/core/callconv.py

```diff
@@ -492,14 +493,12 @@ def unpack_dynamic_exception(self, builder, pyapi, status):
 
         # We check at this stage if creating the Python tuple was successful
         # or not. Note the exception is raised by calling PyErr_SetString
-        # directly as the function being processed is the CPython wrapper
-        # to-do: fix!
-        # failed = cgutils.is_null(builder, py_tuple)
-        # with cgutils.if_unlikely(builder, failed):
-        #     msg = ('Error creating Python tuple from runtime exception '
-        #            'arguments')
-        #     pyapi.err_set_string("PyExc_RuntimeError", msg)
-        #     builder.ret(pyapi.get_null_object())
+        # directly as the current function is the CPython wrapper.
+        failed = cgutils.is_null(builder, py_tuple)
+        with cgutils.if_unlikely(builder, failed):
+            msg = ('Error creating Python tuple from runtime exception '
+                   'arguments')
+            pyapi.err_set_string("PyExc_RuntimeError", msg)
```

Comment:
```suggestion
            pyapi.err_set_string("PyExc_RuntimeError", msg)
            builder.ret(pyapi.get_null_object())
```
I'm not sure that this has any effect unless the error state is communicated through a return value of NULL.